lib/internal/bootstrap/pre_execution.js

```diff
@@ -164,6 +166,19 @@ function initializeReportSignalHandlers() {
   addSignalHandler();
 }
 
+function initializeHeapSnapshotSignalHandlers() {
+  const signal = getOptionValue('--heapsnapshot-signal');
+
+  if (!signal)
+    return;
+
+  const { writeHeapSnapshot } = require('v8');
+
+  process.on(signal, () => {
```

Comment:
This needs to be validated, it could be e.g. `warning`. See `isSignal` in `main_thread_only.js` for an example.

Reply:
I added validation. Once this lands, I'm going to add a centralized signal name validator.